File: The k-th Lexicographical String of All Happy Strings of Length n.py
# https://leetcode.com/problems/the-k-th-lexicographical-string-of-all-happy-strings-of-length-n/?envType=daily-question&envId=2025-02-19
# The k-th Lexicographical String of All Happy Strings of Length n

class Solution:
    def getHappyString(self, n: int, k: int) -> str:
        
        happy_strings = []
        
        def backtrack(path: str):
            if len(path) == n:
                happy_strings.append(path)
                return
            for ch in "abc":
                if not path or path[-1] != ch:
                    backtrack(path + ch)
        
        backtrack("")
        
        happy_strings.sort()
        
        return happy_strings[k - 1] if k <= len(happy_strings) else ""

File: The k-th Lexicographical String of All Happy Strings of Length n.js
// https://leetcode.com/problems/the-k-th-lexicographical-string-of-all-happy-strings-of-length-n/?envType=daily-question&envId=2026-03-14
// The k-th Lexicographical String of All Happy Strings of Length n

/**
 * @param {number} n
 * @param {number} k
 * @return {string}
 */
var getHappyString = function(n, k) {
    const happyStrings = [];

    function backtrack(path) {
        if (path.length === n) {
            happyStrings.push(path);
            return;
        }

        for (const ch of "abc") {
            if (path.length === 0 || path[path.length - 1] !== ch) {
                backtrack(path + ch);
            }
        }
    }

    backtrack("");

    happyStrings.sort();

    return k <= happyStrings.length ? happyStrings[k - 1] : "";
};